var/spack/repos/builtin/packages/proj/package.py

```diff
@@ -83,12 +84,45 @@ class Proj(AutotoolsPackage):
     )
 
     # https://proj.org/install.html#build-requirements
-    depends_on("pkgconfig@0.9.0:", type="build", when="@6:")
     depends_on("googletest", when="@6:")
     depends_on("sqlite@3.11:", when="@6:")
     depends_on("libtiff@4.0:", when="@7:+tiff")
     depends_on("curl@7.29.0:", when="@7:+curl")
+    depends_on("pkgconfig@0.9.0:", type="build", when="@6: build_system=autotools")
+    depends_on("cmake@2.6.0:", type="build", when="build_system=cmake")
 
+    build_system("autotools", conditional("cmake", when="@5.0.0:"), default="cmake")
+
+    def setup_run_environment(self, env):
+        # PROJ_LIB doesn't need to be set. However, it may be set by conda.
+        # If an incompatible version of PROJ is found in PROJ_LIB, it can
+        # cause the package to fail at run-time. See the following for details:
+        # * https://proj.org/usage/environmentvars.html
+        # * https://rasterio.readthedocs.io/en/latest/faq.html
+        env.set("PROJ_LIB", self.prefix.share.proj)
+
+
+class Setup:
+    def setup_dependent_build_environment(self, env, dependent_spec):
+        self.pkg.setup_run_environment(env)
+
+    def setup_dependent_run_environment(self, env, dependent_spec):
+        self.pkg.setup_run_environment(env)
+
+
+class CMakeBuilder(cmake.CMakeBuilder, Setup):
+    def cmake_args(self):
+        args = [
+            self.define("PROJ_LIB", join_path(self.stage.source_path, "nad")),
```

Comment:
This doesn't seem to be a valid cmake flag. We could try setting it as an env var in `setup_build_environment` for both Autotools and CMake to see if that works.

Reply:
Whoops, good catch. Looking at the notes for this port, I grepped for that substring in the cmake options and got it, but looking more closely now the string I actually matched was "PROJ_LIB_SUBDIR"... I'll add to the env.

Reply:
Actually on further inspection we already set that value during both the build and the run in the env. It's set to a different path of course. Seems like this argument in configure is pointing the build to the spack installed resources. I'm sure there must be a corresponding CMake arg, but I'm unsure what this data represents so its difficult to determine which is the correct one. Do you have any further insight there?

Reply:
We aren't setting it in `setup_build_environment`

Reply:
... 🤦 ... good point. I've made that change.